File: samples/scala/demo/app/DemoModule.scala

import play.api.{ Configuration, Environment }
import play.api.inject.{ Binding, Module }
import securesocial.core.RuntimeEnvironment
import service.MyEnvironment

class DemoModule extends Module {
  override def bindings(environment: Environment, configuration: Configuration): Seq[Binding[_]] = {
    Seq(
      bind[RuntimeEnvironment].to[MyEnvironment])
  }
}

File: samples/java/demo/app/DemoModule.java
import com.google.inject.AbstractModule;
import securesocial.core.RuntimeEnvironment;
import service.MyEnvironment;


public class DemoModule extends AbstractModule {
    @Override
    protected void configure() {
        MyEnvironment environment = new MyEnvironment();
        bind(RuntimeEnvironment.class).toInstance(environment);
    }
}
